app/services/wakes/all_pageviews_count_updater_service.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+require 'google/apis/analytics_v3'
+
+class Wakes::AllPageviewsCountUpdaterService
+  def initialize(start_year, end_year = current_year)
+    @start_year = start_year
+    @end_year = end_year
+  end
+
+  def update_path_counts
+    date_ranges.each do |date_range|
+      date_range.update_path_counts.each { |path, count| update_path_count(path, count) }
+    end
+  end
+
+  private
+
+  attr_reader :start_year, :end_year
+
+  def current_year
+    Date.today.year
+  end
+
+  def date_ranges
+    @date_ranges ||= begin
+      start_date = Date.new(start_year, 1, 1)
+      end_date = current_year == end_year ? 1.day.ago.to_date : Date.new(end_year, 12, 31)
+      DateRanges.new(start_date..end_date, :year)
+    end
+  end
+
+  def update_path_count(path, count)
+    location = Wakes::Location.where(:path => path).first
+    return unless location.present?
+
+    location.pageview_counts[@year] = count
+    location.save
+    location.resource&.update_pageview_count
+    begin
+      location.resource.wakeable&.update_pageview_count
+    rescue NameError => ex # Invalid wakeable
+      puts ex.message
+    end
+  end
+
+  class DateRanges
```

Comment:
@joeosburn-desiringgod your class name choices in here are a bit non-standard and a bit confusing...

1. I assume that `DateRange` means "a range of dates". "range" is a technical term in ruby and "date" is a technical term in English and in ruby. 
2. I _think_ by "date range" you mean "all the pageviews for a date range", right?
3. In line 57, you use a `GetDateRange` class to assign a `date_range` instance variable. Why not have a `DateRange` class? Why is `Get` necessary?

A related thing is that you are doing a _lot_ more in these classes than I would have suspected based on their names. Better names would help, as would pulling some actions out into the primary `update_path_counts` method at line 10.

Here's what makes sense to me/how I expected it to work upon first scanning it:

1. `Wakes::AllPageviewCountUpdaterService#date_ranges` returns an array of `DateRange`, which is an object that basically just has a start date and an end date.
2. That `date_range` is then passed into a new `Wakes::AllPageviewCountUpdaterService::GetPathCountsForDateRangeService` object, which does the heavy lifting that you are doing right now in `GetDateRange`

I'm sure there are some details I'm missing here, but that's a broad-brush attempt.